apps/ui/src/pages/manage/applications/index.tsx

```diff
@@ -0,0 +1,335 @@
+ * @param column The column object passed in by the `antd` table.
+ * @returns A `ApplicationListSortingOptions` compatible column name.
+ */
+const parseSortingOptions = (column: string | Key | readonly Key[]) => {
+	let key = column;
+	if (Array.isArray(column)) {
+		key = column[column.length - 1];
+	}
+
+	switch (key) {
+		case 'id':
+			return 'id';
+		case 'applicant':
+			return 'user_id';
+		case 'createdAt':
+			return 'created_at';
+		case 'approvedAt':
+			return 'approved_at';
+		case 'updatedAt':
+			return 'updated_at';
+		case 'expiresAt':
+			return 'expires_at';
+		case 'state':
+			return 'state';
+		default:
+			return 'id';
+	}
+};
+
+/**
+ * Calculates the number shown beside the filter at the top of the page.
+ * Example: 10 Total | 3 DAC Review, etc...
+ * @param data The data provided via the backend `metadata/counts` endpoint.
+ * @returns A `FilterState` object, containing the unique key of the filter and how many applications are filed under it.
+ */
+const calculateFilterAmounts = (countMetadata: ApplicationCountMetadata) => {
+	const availableStates: FilterState[] = [];
+
+	for (const appState of Object.keys(countMetadata)) {
+		availableStates.push({
+			key: appState as ApplicationStateValues,
+			amount: countMetadata[appState as keyof ApplicationCountMetadata],
+		});
+	}
+	return availableStates;
+};
+
+/**
+ * Processes a page number passed into it, checking if it's valid, and if it needs to be converted into a int.
+ *
+ * Additionally, since page numbers on the server count from 0, whereas `antd` counts from 1, it can convert from
+ * either or using the `forNetwork` param.
+ * @param pageNumber The current page that the user is on in the UI from the URL
+ * @param forNetwork If the processed page number you'd like back is being sent to the server (count from zero)
+ * @returns A `number` that has been validated and converted.
+ */
+const parsePageNumber = (pageNumber?: number | string | null, forNetwork?: boolean) => {
+	if (pageNumber) {
+		const parsedPage = typeof pageNumber === 'string' ? parseInt(pageNumber) : pageNumber;
+
+		if (isValidPageNumber(parsedPage)) {
+			return forNetwork ? parsedPage - 1 : parsedPage;
+		}
+
+		return forNetwork ? 0 : 1;
+	} else {
+		return forNetwork ? 0 : 1;
+	}
+};
+
+/**
+ * Gets and parses filters from the URL params. It also validates if the filters are contained within
+ * the possible set of filters. It then returns the set decoded and ready to use.
+ *
+ * @param filtersFromQuery The raw string of filters from the query param.
+ * @returns A validated and typed set of the current filters. Will return just `TOTAL` if set is invalid.
+ */
+const parseFilters = (filtersFromQuery?: string | null) => {
+	if (filtersFromQuery) {
+		const decodedFilters = filtersFromQuery.split(',');
+		if (isValidFilterSet(decodedFilters)) {
+			return decodedFilters as FilterKeys[];
+		}
+		return ['TOTAL'] as FilterKeys[];
+	} else {
+		return ['TOTAL'] as FilterKeys[];
+	}
```

Comment:
Returning the same thing twice, and using `as` three times vs `const parseFilters = (filtersFromQuery?: string | null): FilterKeys[]` so I'd want to review how we can clear this up